### DISCUSSION TITLE
Get descriptive error messages from the C API?

### DISCUSSION THREAD

**Discussion by wusspuss** [paullouisageneau/libdatachannel#958] [Q&A]
E.g. calling `rtcSetLocalDescription` with an invalid SDP string seems to only return `RTC_ERR_FAILURE`, which is rather ambiguous compared to the error messages I see in the C++ code, like `throw std::logic_error("Illegal role actpass in remote answer description");` Is there currently a way to see those exception messages from C?
I see these errors in logger output, still it would be useful to get the error from the particular line that caused it.

**Comment by paullouisageneau** [comment]
Yes, this could be useful to add a function allowing to access the actual error message from the C API.

As a preliminary step I have changed all exceptions meaning the remote description is invalid to invalid argument in https://github.com/paullouisageneau/libdatachannel/pull/962, which will translate to `RTC_ERR_INVALID`.
